components/dashboard/src/projects/NewProject.tsx

```diff
@@ -69,7 +92,7 @@ export default function NewProject() {
             window.history.replaceState({}, "", window.location.pathname);
             setSelectedTeamOrUser(user);
         }
-    }, []);
+    }, [location.search, teams, user]);
```

Comment:
I'm not 100% sure whether this change makes sense:
- Currently, we only call this effect once, and we immediately "consume" the search parameter if found (i.e. if you visit `/new?team=1234`, we immediately select team `1234` and overwrite the URL to `/new`). This means that, if we call this code multiple times, only the first time can have any effect.
- Calling it again if `location.search` changes sort of makes sense, but can it really happen that `location.search` changes without a reload? (I guess it could happen if we have something like `<Link to="/new?team=1234"/>` 🤔)
- Can `teams` or `user` really change here? If `teams` or `user` can be something unexpected, I guess we'd need to check for that in the beginning of the effect, and return early if the values are unexpected (e.g. undefined or empty array?)

All in all, since this code currently seems to work in all cases, I'm a bit hesitant to change the effect logic here.

Reply:
Thanks! I generally agree with you, but I believe I had one bad experience where I just followed ESLint and it broke my component 😬 (or, maybe the bug was my fault and I now wrongly blame ESLint for it). I guess this calls for more testing. 👀

Reply:
I think it happened to me once!
I don't remember the specifics but like you said turns out it was my "fault" meaning I then applied "more React" to the problem (IIRC more composition and maybe a custom hook 🤔 ) and that fixed both the warning and the bug 🐞